src/dune_rules/pkg_rules.ml

```diff
@@ -144,6 +144,40 @@ module Install_cookie = struct
   ;;
 end
 
+module Typed_env = struct
+  (* A representation of an environment where each variable can hold a
+     list of [Value.t]. Each variable will be encoded into a delimited
+     string (in the style of the PATH variable). *)
+  type t = Value.t list Env.Map.t
+
+  let parse_strings s = Bin.parse s |> List.map ~f:(fun s -> Value.String s)
+  let of_env env : t = Env.to_map env |> Env.Map.map ~f:parse_strings
+
+  let string_of_env_values values =
+    List.map values ~f:(function
+      | Value.String s -> s
+      | Dir s | Path s -> Path.to_absolute_filename s)
+    |> Bin.encode_strings
+  ;;
+
+  let to_env (t : t) = Env.Map.map t ~f:string_of_env_values |> Env.of_map
+  let get_path t = Env.Map.find t Env_path.var
+
+  (* [extend_concat_path a b] adds all variables from [b] to [a]
+     overwriting any existing values of those variables in [a] except for PATH
+     which is set to the concatenation of the PATH variables from [a] and [b]
+     with the PATH entries from [b] preceding the PATH entries from [a] *)
+  let extend_concat_path a b =
+    let path =
+      match get_path a, get_path b with
+      | None, None -> []
```

Comment:
Does this mean if neither side has `PATH` set, you will set it to the empty path?

Reply:
That's correct, though thinking more about this I think what it should be doing instead is not setting `PATH` at all in such a case. Will update.

Reply:
Updated